pymc3/model.py

```diff
@@ -935,50 +943,214 @@ def test_point(self) -> Dict[str, np.ndarray]:
     @property
     def initial_point(self) -> Dict[str, np.ndarray]:
         """Maps free variable names to transformed, numeric initial values."""
-        if set(self._initial_point_cache) != {get_var_name(k) for k in self.initial_values}:
-            return self.recompute_initial_point()
-        return self._initial_point_cache
+        seed = self.rng_seeder.randint(2 ** 30, dtype=np.int64)
+        return self.recompute_initial_point(seed)
+
+    def recompute_initial_point(self, seed) -> Dict[str, np.ndarray]:
+        """Recomputes the initial point of the model.
+
+        Returns
+        -------
+        ip : dict
+            Maps names of transformed variables to numeric initial values in the transformed space.
+        """
+        fn = self.make_initial_point_fn(return_transformed=True)
+        return Point(fn(seed), model=self)
 
-    def recompute_initial_point(self) -> Dict[str, np.ndarray]:
+    def _make_initial_point_expression(
+        self,
+        *,
+        jitter_rvs: Set[TensorVariable] = None,
+        default_strategy: str = "prior",
+        return_transformed: bool = False,
+    ) -> List[TensorVariable]:
         """Recomputes numeric initial values for all free model variables.
 
+        Parameters
+        ----------
+        jitter_rvs : set
+            The set (or list or tuple) of random variables for which a U(-1, +1) jitter should be
+            added to the initial value. Only available for variables that have a transform or real-valued support.
+        default_strategy : str
+            Which of { "moment", "prior" } to prefer if the initval setting for an RV is None.
+        return_transformed : bool
+            Switches between returning the dictionary based on RV vars or RV value vars as keys.
+
+        Returns
+        -------
+        initial_points : list of TensorVariable
+            Aesara expressions for initial values of the free random variables.
+        """
+        from pymc3.distributions.distribution import get_moment
+
+        if jitter_rvs is None:
+            jitter_rvs = set()
+
+        initial_values = []
+        initial_values_transformed = []
+
+        for variable in self.free_RVs:
+            strategy = self.initial_values.get(variable, None)
+            if strategy is None:
+                strategy = default_strategy
```

Comment:
```suggestion
            strategy = self.initial_values.get(variable, default_strategy)
```

Reply:
So far we put `None` in the `self.initial_values` too (to indicate that no choice was made).

Reply:
What do we do with that info?

Reply:
We fall back to the `default_strategy`.
It's a way of enforcing the default strategy and generally more robust, even if we decide to not store `None`-valued initvals, which I would advise against in comparison to a dict that has entries for all model variables.

Reply:
My suggestion line does exactly the same, but in a single line, that's all

Reply:
No it doesn't: If there's a `None`-valued entry in `initial_values`, the `get` method will return the `None`, not the `default_strategy`

Reply:
Maybe I am looking at an outdated version, but when I commented there was an immediate follow-up, which is exactly the same.
```
strategy = self.initial_values.get(variable, None)
if strategy is None:
   strategy = default_strategy
```

Reply:
see new thread